universal-application-tool-0.0.1/app/auth/saml/SamlCiviFormProfileAdapter.java

```diff
@@ -155,6 +182,13 @@ public CiviFormProfileData mergeCiviFormProfile(
     }
     String emailAddress = saml2Profile.getEmail();
     civiFormProfile.setEmailAddress(emailAddress).join();
+
+    Optional<String> authorityId = getAuthorityId(saml2Profile);
+    if (authorityId.isEmpty()) {
+      throw new InvalidSamlProfileException("Unable to get authority ID from profile.");
+    }
+    civiFormProfile.setAuthorityId(authorityId.get()).join();
```

Comment:
Since we never had email as a key in SAML, can you make setting the authority the first one we set and email second? 

As is we could create a record with the email then throw an exception getting/setting the authority.

Reply:
Fixed. Should I make setting authority ID the very first thing I do, then throw an exception if that couldn't be found? IE, before setting locale/ name?

Reply:
I'm not sure why locale/name came first before email in the previous code...

Those look to be set on Applicant though, and these are on Account so I guess it may not matter which is first.

Reply:
Moved it to the top! This definitely seems like the best way